### PR TITLE
Restored Smarty legacy functions for pdf

### PR DESCRIPTION
**Pull request by julienbourdeau**
| Questions | Answers |
| --- | --- |
| Branch? | develop |
| Description? | PDF still use smarty function like `{displayPrice xx}` so many smarty function have been restored |
| Type? | bug fix |
| Category? | FO |
| BC breaks? | no |
| Deprecations? | no |
| Fixed ticket? | http://forge.prestashop.com/browse/BOOM-1276 |
| How to test? |  |

### CODE REVIEW

**Comment by vincentbz** [comment]
Tested and ok, thanks !

File: classes/controller/PdfFrontController.php [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<?php
+
```
Comment (mickaelandrieu) [inline classes/controller/PdfFrontController.php]: remove lines please :)

File: classes/controller/PdfFrontController.php [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+abstract class PdfFrontControllerCore extends FrontController
+{
+    public function init()
+    {
+        parent::init();
+
+        /* For PDF we restore some function from Smarty
+         * they've been removed in PrestaShop 1.7 so
+         * new themes don't them. Although PDF haven't been
```
Comment (mickaelandrieu) [inline classes/controller/PdfFrontController.php]:
```diff
- don't them
+ don't have them
```

File: classes/controller/PdfFrontController.php [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+         * rework so every PDF controller must extend this class.
```
Comment (mickaelandrieu) [inline classes/controller/PdfFrontController.php]:
```diff
- rework
+ reworked
```

**Comment by mickaelandrieu** [comment]
Some little comments to be adressed, good job @julienbourdeau !

File: classes/controller/PdfFrontController.php [hunk 4]
```diff
@@ -0,0 +1,25 @@
+<?php
+
+
+abstract class PdfFrontControllerCore extends FrontController
+{
+    public function init()
+    {
+        parent::init();
+
+        /* For PDF we restore some function from Smarty
```
Comment (xBorderie) [inline classes/controller/PdfFrontController.php]: functionS ;)

**Comment by mickaelandrieu** [comment]
Thank you @julienbourdeau

**Pull request merged**